### PR TITLE
feat: adjust Dialog components, introduce DialogIcon

### PR DESCRIPTION
**Pull request by lukewalczak**
### Summary

PR:
* adjusts `Dialog` components to Material You
* introduces `DialogIcon` component

### CODE REVIEW

File: src/components/Dialog/DialogTitle.tsx [hunk 1]
```diff
@@ -50,21 +51,25 @@ type Props = React.ComponentPropsWithRef<typeof Title> & {
...
+const DialogTitle = ({ children, theme, style, ...rest }: Props) => {
+  const { isV3 } = theme;
+
+  const TextComponent = isV3 ? Text : Title;
+  const textColor = isV3 ? theme.colors.onSurface : theme.colors?.text;
+
+  return (
+    <TextComponent
+      variant="headlineSmall"
+      // @ts-expect-error We keep old a11y props for backwards compat with old RN versions
```
Comment (jaworek) [inline src/components/Dialog/DialogTitle.tsx:63]: Do we need to keep this? New version will include some breaking changes so maybe this is a good opportunity to stop supporting older RN versions? Although, it depends on how old they are.

File: example/src/Examples/Dialogs/utils.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import type { MD3TypescaleKey } from '../../../../src/types';
+
+type Props = React.ComponentProps<typeof NativeText> & {
+  style?: StyleProp<TextStyle>;
+  variant?: keyof typeof MD3TypescaleKey;
+  children: React.ReactNode;
+  isSubheading?: boolean;
+};
+
+export const TextComponent = ({ isSubheading = false, ...props }: Props) => {
```
Comment (jaworek) [inline example/src/Examples/Dialogs/utils.tsx:19]: I think it would be better to name this file the same as component, unless you're planning to have more utils. Although, in that case you could have a folder named `utils` with each util in a separate file.

File: example/src/Examples/Dialogs/DialogWithLongText.tsx [hunk 3]
```diff
@@ -56,7 +57,7 @@ const DialogWithLongText = ({
             conveying how objects move, interact, and exist in space and in
             relation to each other. Realistic lighting shows seams, divides
             space, and indicates moving parts.
-          </Paragraph>
+          </TextComponent>
         </ScrollView>
       </Dialog.ScrollArea>
       <Dialog.Actions>
```
Comment (jaworek) [inline example/src/Examples/Dialogs/DialogWithLongText.tsx:63]: I noticed small inconsistency that is not related to this PR. All Dialogs use uppercase first letter and rest lowercase. Long text dialog is the only one that has "OK" instead of "Ok".

File: src/components/Dialog/DialogTitle.tsx (hunk 1 above)
Comment (lukewalczak) [reply src/components/Dialog/DialogTitle.tsx:63]: The plan is to remove it along with the new version, however, it will be tracked in the separated PR.

File: example/src/Examples/Dialogs/utils.tsx (hunk 2 above)
Comment (lukewalczak) [reply example/src/Examples/Dialogs/utils.tsx]: Yup, I've renamed it, since actually, it's just a component instead of util.

**Review by jaworek — APPROVED** [review]

File: src/components/Dialog/DialogTitle.tsx (hunk 1 above)
Comment (lukewalczak) [reply src/components/Dialog/DialogTitle.tsx:63]: https://github.com/callstack/react-native-paper/pull/2484

**Review by kwereszczynski — APPROVED** [review]
Looks good to me :)
